build/dsc/templates/dsc_type.rb.erb

```diff
@@ -86,11 +86,15 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
 
   ensurable do
     newvalue(:exists?) { provider.exists? }
-    newvalue(:<%= begin resource.ensure_value rescue 'present' end %>) { provider.create }
+<%  resource.valid_ensure_present_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>) { provider.create }
+<%  end -%>
 <%  if resource.absentable? -%>
-    newvalue(:<%= resource.absent_value %>)  { provider.destroy }
+<%  resource.valid_ensure_absent_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>)  { provider.destroy }
```

Comment:
Yup, understand the symmetry argument - it's just that I'm not sure it's something we need to support, because I don't think there would ever be multiple values that *should* trigger `provider.destroy`.  Even with AWS EC2 instances, that have the most `ensure` values that I've seen (`present`, `absent`, `running`, `stopped`), only `absent` calls `destroy` because `stopped` has a different semantic meaning - see code at https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet/type/ec2_instance.rb#L6-L31

Based on how we've implemented the bridge here, I think we can only really call `provider.create` for anything other than `absent`